src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -299,10 +299,15 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
 
   double performance = 0;
   size_t responseSeq = 0;
+  size_t r_size = responses.size();
```

Comment:
What I think is, we could avoid introducing new variable , and change the loop to -

`for (size_t seqNum = 0; seqNum < min(rho, responses.size()); ++seqNum)`
 
not sure :)

Reply:
Their types are not matching 
```note:   mismatched types ‘std::initializer_list<_Tp>’ and ‘long unsigned int’
   for (size_t seqNum = 0; seqNum < std::min(rho, responses.size()); ++seqNum)```

Reply:
@himanshupathak21061998 what have you tried to debug the issue?  I agree with @jeffin143 that we can avoid the need for `r_size` and we should.  Even just doing `const size_t effectiveRho = std::min(rho, responses.size())` (or something like it) would be an improvement in my view.